src/map.cpp

```diff
@@ -4729,11 +4729,11 @@ item &map::add_item_at( const tripoint &p,
 item map::water_from(const tripoint &p)
 {
     if( has_flag( "SALT_WATER", p ) ) {
-        item ret( "salt_water", 0 );
+        item ret( "salt_water", 0, std::numeric_limits<long>::max() );
         return ret;
     }
 
-    item ret( "water", 0 );
+    item ret( "water", 0, std::numeric_limits<long>::max() );
```

Comment:
The `item` constructor appears to be at fault here.

We should probably define `typedef long charges_type` and gradually start migrating towards using this with the eventual goal of expressing `charges` as an `int`